### PR TITLE
refactor: move encodings to lance-core

### PR DESCRIPTION
**Pull request by eddyxu**

### CODE REVIEW

File: rust/lance-core/src/encodings/plain.rs [hunk 1]
```diff
@@ -748,198 +741,121 @@ mod tests {
...
-
-        let reader = FileReader::try_new(&store, &path).await.unwrap();
-        let actual = reader.take(&[2, 4, 5, 8, 4555], &schema).await.unwrap();
-
-        assert_eq!(
-            actual.column_by_name("b").unwrap().as_ref(),
-            &BooleanArray::from(vec![false, false, true, false, true])
-        );
-    }
+    // #[tokio::test]
```
Comment (chebbyChefNEQ) [inline rust/lance-core/src/encodings/plain.rs:746]: why are these disabled?

Comment (eddyxu) [reply rust/lance-core/src/encodings/plain.rs:746]: This requires `FileReader / FileWriter`, which have not been migrated yet.

Comment (chebbyChefNEQ) [reply rust/lance-core/src/encodings/plain.rs:746]: I guess we need to finish the migration before next release?

**Review by chebbyChefNEQ — APPROVED** [review]

File: rust/lance-core/src/encodings/plain.rs (hunk 1 above)
Comment (eddyxu) [reply rust/lance-core/src/encodings/plain.rs:746]: it is ok, no user facing changes.  The pub structure / traits are re-exported in `lance`

**Pull request merged**